aim2dat/strct/ext_manipulation/add_functional_group.py

```diff
@@ -115,20 +114,15 @@ def add_functional_group(
         rot_matrix = rotation.as_matrix()
 
         # Create updated structure:
-        new_structure = {
-            "label": structure["label"],
-            "pbc": structure["pbc"],
-            "is_cartesian": True,
-            "wrap": wrap,
-            "positions": list(structure["positions"]),
-            "cell": structure["cell"],
-            "elements": list(structure["elements"]),
-        }
+        new_structure = structure.to_dict()
+        new_structure["elements"] = list(new_structure["elements"])
+        new_structure["positions"] = list(new_structure["positions"])
```

Comment:
`structure.to_dict()` also contains the `kinds` attribute. Therefore, we are causing an inconsistency between the `elements` and `kinds` in this case. This problem wasn't present before, as the dictionary was initialized from scratch.

Reply:
Thanks and I agree, I have implemented a fix and adapted the test case.